### ISSUE TITLE
Gradient Checkpointing Error

### ISSUE THREAD

**Issue by shenoynikhil** [torchmd/torchmd-net#310]
I was trying to use gradient checkpointing with TorchMD model. For some reason, I get this gradient mismatch whenever the warning pops up 

https://github.com/torchmd/torchmd-net/blob/8a1be712a397d88eed82ae12066f1533f12152ee/torchmdnet/models/utils.py#L492-L499

I am guessing because the activations that are stored do not match with the weight shapes. I modified the code to be like,
```
        mask = (vec1_buffer != 0).view(vec1_buffer.size(0), -1).all(dim=1)
        vec1 = torch.where(
            mask.unsqueeze(1), # where to apply the mask
            torch.norm(vec1_buffer, dim=-2), # On non-zero values, take norm
            torch.sum(vec1_buffer, dim=-2) # On 0s, keeping this instead of new tensor for gradients
        )
```
This seems to subvert the error.

**Comment by RaulPPelaez** [comment]
I believe this should be ok to ignore if you are ok with its message (a.i. you know the cutoff is such that some atoms are left alone). If you know for a fact this is not the case but the message pops up anyway then this is a bug.
